sql-api/src/main/java/io/cdap/pipeline/sql/api/template/interfaces/SQLNode.java

```diff
@@ -30,15 +32,19 @@
    * This method should be reserved for logic which needs the overall pipeline context (e.g. joins). For
    * single-input transformations (e.g. aggregation, projection, or filter), use the getQuery() method.
    *
+   * combineInputs is guaranteed to be called exactly once prior to getQuery.
+   *
    * @param builder The {@link RelNode} expression builder used to generate the query
-   * @param dependencies The number of input nodes which were passed in
+   * @param inputs A list of input stage names
    * @return The relational expression builder with one input on the stack
    */
-  RelBuilder combineInputs(RelBuilder builder, int dependencies);
+  RelBuilder combineInputs(RelBuilder builder, List<String> inputs);
```

Comment:
I don't think this method should exist. It shouldn't be up to the plugin to decide how to combine inputs. Rather, the app should enforce that multiple inputs always mean a union is being performed, unless dealing with a joiner. 

When dealing with a joiner, the app should enforce that there are exactly 2 inputs, and leave it up to the plugin to do the right thing in getQuery().

Reply:
Fixed